### ISSUE TITLE
ImageBitmapLoader Cache key should take account of ImageBitmapOptions

### ISSUE THREAD

**Issue by takahirox** [mrdoob/three.js#16196]
##### Description of the problem

**Background**

`ImageBitmap` is created with `createImageBitmap()`. `createImageBitmap()` can take options `imageOrientation(flipY)`, `premultiplyAlpha`, and so on. The options have effect to bitmap.

https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/createImageBitmap

So `ImageBitmap` created with different options parameters should be distinguished from others even if they refer to the same url.

**Problem**

But `Cache` system in `ImageBitmapLoader` uses only url as cache key.

https://github.com/mrdoob/three.js/blob/dev/src/loaders/ImageBitmapLoader.js#L50
https://github.com/mrdoob/three.js/blob/dev/src/loaders/ImageBitmapLoader.js#L87

In the example below, `bitmap` should be distinguished from `bitmap2` but identical if `Cache` is enabled.

```javascript
var loader = new THREE.ImageBitmapLoader();

loader.setOptions( { imageOrientation: 'none' } );
loader.load( url, function ( bitmap ) {

    loader.setOptions( { imageOrientation: 'flipY' } );
    loader.load( url, function ( bitmap2 ) {

        console.log( bitmap === bitmap2 ); // Should be false

    } );

} );
```

##### Three.js version

- [x] Dev
- [ ] r103
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by Mugen87** [comment]
> So ImageBitmap created with different options parameters should be distinguished from others even if they refer to the same url.

TBH, this use case sounds a bit artificial to me. Why would you load the same texture twice with different image bitmap settings? Properties like `premultiplyAlpha` or `colorSpaceConversion` highly depend on the used image. It's seems unlikely to extract the same image data in different ways...

**Comment by takahirox** [comment]
Yeah, may be rare case. But it's a bug. Probably accepting it's as limitation and documenting may sound reasonable.
